tests/e2e/unified_test.go

```diff
@@ -273,11 +274,42 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 				}
 				create.RunCreateDeleteTest(h, opt)
 
-				for _, obj := range exportResources {
-					got := exportResource(h, obj)
-					if got != "" {
-						expectedPath := filepath.Join(fixture.SourceDir, "export.yaml")
-						h.CompareGoldenFile(expectedPath, string(got), IgnoreComments, ReplaceString(project.ProjectID, "example-project-id"))
+				if os.Getenv("GOLDEN_OBJECT_CHECKS") != "" {
+					for _, obj := range exportResources {
+						// Construct file path to save golden files
+						dir := "golden/"
+						pieces := strings.Split(t.Name(), "/")
+						var testName string
+						if len(pieces) > 0 {
+							testName = pieces[len(pieces)-1]
+						} else {
+							t.Errorf("failed to get test name")
+						}
+						// Golden test exported GCP object
+						got := exportResource(h, obj)
+						if got != "" {
+							expectedPath := filepath.Join(dir, fmt.Sprintf("export/_%v.yaml", testName))
+							h.CompareGoldenFile(expectedPath, string(got), IgnoreComments, ReplaceString(project.ProjectID, "example-project-id"))
+						}
+						// Golden test KRM object schema
+						u, err := getKRMObject(h, obj)
+						if u == nil {
+							t.Errorf("failed to get KRM object: %v", err)
+						} else {
+							if err := normalizeObject(u, project, uniqueID); err != nil {
+								t.Fatalf("error from normalizeObject: %v", err)
+							}
+							got, err := yaml.Marshal(u)
+							if err != nil {
+								t.Errorf("failed to convert KRM object to yaml: %v", err)
+							}
+							// File name looks like _testname.yaml
+							expectedPath := filepath.Join(dir, fmt.Sprintf("object/_%v.yaml", testName))
```

Comment:
Ditto here, fixture.SourceDir feels more natural to me - WDYT?

Reply:
I'm curious, why is it `natural`? Is there any design pattern or best practice related to it? My intuition is actually more aligned with @gemmahou 's design choice so would like to learn more about it.